carbonserver/carbonserver/api/infra/database/sql_models.py

```diff
@@ -177,17 +177,22 @@ class ProjectToken(Base):
     id = Column(UUID(as_uuid=True), primary_key=True, index=True, default=uuid.uuid4)
     project_id = Column(UUID(as_uuid=True), ForeignKey("projects.id"))
     name = Column(String)
-    token = Column(String, unique=True)
+    hashed_token = Column(String,nullable=False)
+    lookup_value = Column(String,nullable=False) # This is the first 8 characters of the SHA-256 hash of the API key. Used for filtering faster
+    revoked = Column(Boolean, default=False)
     project = relationship("Project", back_populates="project_tokens")
+    # Dates
     last_used = Column(DateTime, nullable=True)
+    expiration_date = Column(DateTime, nullable=False)
```

Comment:
lets not handle token validity by date, we have no need for this complexity

Reply:
Removed! See [fe6dcfe](https://github.com/mlco2/codecarbon/pull/688/commits/fe6dcfe13b45a9915d8990a0fc780e4fc00e913f)